src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.WorkItem.cs

```diff
@@ -106,14 +106,10 @@ public WorkItem(DocumentId documentId, string language, InvocationReasons invoca
 
                 private ImmutableHashSet<IIncrementalAnalyzer> Union(ImmutableHashSet<IIncrementalAnalyzer> analyzers)
                 {
-                    if (analyzers.IsEmpty)
+                    // An empty analyzer list means run all analyzers, so empty always wins
```

Comment:
The logic in this method no longer seems to indicate a Union operation. I think it would be better to make this a static method with following signature:

```csharp
private static ImmutableHashSet<IIncrementalAnalyzer> ComputeNewSpecificAnalyzers(ImmutableHashSet<IIncrementalAnalyzer> specificAnalyzers1, ImmutableHashSet<IIncrementalAnalyzer> specificAnalyzers2)
```

Reply:
Also, can we make this a local function?